internal/backend/temporalclient/largepayloads_codec.go

```diff
@@ -0,0 +1,178 @@
+package temporalclient
+
+import (
+	"errors"
+	"maps"
+	"path/filepath"
+	"strconv"
+
+	"github.com/philippgille/gokv"
```

Comment:
A KV abstraction is an unnecessary layer for this feature, it just adds complication and risk.

All we need is S3, and in the distant future we'll see if/how we need to support other clouds based on business needs.

Moreover, this specific package doesn't seem to be actively maintained, or even that popular. Even worse, it's so old that it uses insecure crypto deps (see unreviewed bot PRs), and an ancient AWS package instead of v2 which we already depend on.

Reply:
Disagree - this is a very simple package, merely an interface with some implementations. This allows also to work locally with blobs.

Reply:
brought kv in and redid the s3